emojirades/slack.py

```diff
@@ -40,12 +48,7 @@ def get_names(self, user_id):
         }
 
     def pretty_name(self, user_id):
-        if user_id not in self.pretty_names:
-            self.pretty_names[user_id] = self.user_info(user_id)
-
-        user = self.pretty_names[user_id]
-
-        return user.get("real_name", user.get("name", "Unknown User"))
+        return self.user_info(user_id).get("real_name", user.get("name", "Unknown User"))
```

Comment:
I guess the second `user.get` is supposed to be `self.user_info(user_id).get`? also can be cached to `user = self.user_info(user_id)`. Perhaps a `with self.user_info(user_id) as user:`.. or `:=` operator somewhere...  endless possibilities and I should probably stop talking (typing) now.

Reply:
Ohh shit good point, logic bug! As I removed the 'user' and this would have triggered an exception.